src/artwork.c

```diff
@@ -971,6 +1025,65 @@ artwork_get_item_path(char *path, int artwork, int max_w, int max_h, struct evbu
 }
 
 /*
+ * Get the artwork for the given media file and the given maxiumum width/height
+
+ * @param mfi the media file structure for the file whose image should be returned
+ * @param max_w maximum image width
+ * @param max_h maximum image height
+ * @param evbuf the event buffer that will contain the (scaled) image
+ * @return 0 or the format code for the returned image on success, -1 on failure
+ */
+static int
+artwork_get_item_mfi(struct media_file_info *mfi, int max_w, int max_h, struct evbuffer *evbuf)
+{
+    int ret;
+    int cached;
+    int format;
+    char filename[PATH_MAX];
+
+    DPRINTF(E_DBG, L_ART, "Artwork request for item %d\n", mfi->id);
+
+    ret = 0;
+    format = 0;
+
+    ret = cache_artwork_get(CACHE_ARTWORK_INDIVIDUAL, mfi->id, max_w, max_h, &cached, &format, evbuf);
+    if (ret == 0 && cached)
+      {
+        if (format > 0)
+  	{
+  	  // Artwork found in cache "ret" contains the format of the image
+  	  DPRINTF(E_DBG, L_ART, "Artwork found in cache for item %d\n", mfi->id);
+  	  return format;
+  	}
+        else if (format == 0)
+  	{
+  	  // Entry found in cache but there is not artwork available
+  	  DPRINTF(E_DBG, L_ART, "Artwork found in cache but no image available for item %d\n", mfi->id);
+  	  return -1;
+  	}
+      }
+
+    if (mfi->data_kind == 0)
+      {
+	if (mfi->artwork == ARTWORK_OWN || mfi->artwork == ARTWORK_UNKNOWN)
```

Comment:
Question regarding this part: Did you consider putting artwork_get_own_image inside artwork_get_item_path? So just calling artwork_get_item_path here, and then in there have something like:
if (individual artwork configured)
case ARTWORK_OWN:
case ARTWORK_UNKNOWN:
artwork_get_own_image(...)

This should have the effect that individual artwork would also be used as group artwork, if nothing else is available - wouldn't that be desirable? Maybe it messes something else up I didn't think of?